envi/archs/arm/disasm.py

```diff
@@ -1773,20 +1837,23 @@ def render(self, mcanv, op, idx):
         mcanv.addNameText(rname, typename='registers')
         if self.shimm != 0:
             mcanv.addText(', ')
-            mcanv.addNameText(shift_names[self.shtype])
-            mcanv.addText(' ')
+            if self.display_shift:
+                mcanv.addNameText(shift_names[self.shtype])
+                mcanv.addText(' ')
             mcanv.addNameText('#%d' % self.shimm)
-        elif self.shtype == S_RRX:
+        elif self.shtype == S_RRX and self.display_shift:
```

Comment:
if we don't keep this functionality, please remove the changes wherever they are. :)